app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt

```diff
@@ -97,19 +110,81 @@ class AccountSettingsFragment : PreferenceFragmentCompat(), CoroutineScope {
         }
 
         // Sync stopped successfully.
-        override fun onIdle() {
+        override fun onIdle(result: SyncResult) {
             CoroutineScope(Dispatchers.Main).launch {
                 val pref = findPreference<Preference>(context!!.getPreferenceKey(R.string.pref_key_sync_now))
                 pref?.let {
                     pref.title = getString(R.string.preferences_sync_now)
                     pref.isEnabled = true
                     updateLastSyncedTimePref(context!!, pref, failed = false)
                 }
+
+                Log.i(LOGTAG, "Sync finished; recording telemetry")
+                for ((_, status) in result) {
+                    status.stats?.let {
+                        it.syncs.forEach {
+                            it.engines.forEach {
+                                when (it.name) {
+                                    "history" -> {
+                                        Log.i(LOGTAG, "Got history sync")
+                                        HistorySync.apply {
+                                            startedAt.set(Date(it.at.toLong()))
+                                            // Glean doesn't support recording arbitrary durations in
```

Comment:
Hi Lina, I think we would rather look at your use case for recording an arbitrary duration rather than modifying the `CounterMetricType`'s specification to fit this use case.  

First let me ask what may be an obvious question: does the sync engine have the capability to interact with a `TimespanMetricType` directly (i.e. to call the `start()` and `stop()` functions?  If it can, then that would be ideal as all of the timing details are handled internally by Glean.  If it cannot, then maybe we need to examine other alternatives such as exposing a public API function to the Timespan that allows a more direct `record()` type of functionality rather than shoehorning a Counter metric to do it.  I'm not sure what your schedule or timezone is, but I'd be willing to bet any one of us on the Glean team would be happy to work this out with you.  @Dexterp37, @mdboom, or myself.

Reply:
> First let me ask what may be an obvious question: does the sync engine have the capability to interact with a `TimespanMetricType` directly (i.e. to call the `start()` and `stop()` functions?

Not directly. The Sync engine is implemented in Rust, and doesn't have access to Glean at all. IIUC, to use Glean directly, it would need to use JNA to call into Fenix (unless we could add the engine pings to a-c...which would be less complicated, but we'd really prefer to avoid that). To my knowledge, Glean isn't an option on iOS at all, so we still use the legacy submission endpoint to send Sync pings there.

Instead, we've gone with the approach of having the Rust sync code build and return a Sync ping, and expose that over its FFI. Firefox for iOS will take that JSON ping and submit it directly (one ping per sync); Fenix unpacks it into Glean pings, and submits them (one ping per engine per sync).

> I apologize, for whatever reason I saw the changes you made to the `CounterMetricType` here and thought that was the type you were using.

Ahhh, yep, that's a different change—I did that because, if a counter for any of the engines already has a value, we want to overwrite it, not add to it, or we'll get wrong counts for each sync. OTOH, if all counters get reset to 0 after calling `Pings.historySync.send()` or `Pings.bookmarksSync.send()` (even if the ping fails to submit), maybe I can drop that and just use `add()`?

Reply:
> If it cannot, then maybe we need to examine other alternatives such as exposing a public API function to the Timespan that allows a more direct `record()` type of functionality...

This would still be really useful to have!

Reply:
> [...] To my knowledge, Glean isn't an option on iOS at all, so we still use the legacy submission endpoint to send Sync pings there.

This is going to change :) We're working hard to have Glean for iOS somewhen in Q3.

> > I apologize, for whatever reason I saw the changes you made to the `CounterMetricType` here and thought that was the type you were using.
> 
> Ahhh, yep, that's a different change—I did that because, if a counter for any of the engines already has a value, we want to overwrite it, not add to it, or we'll get wrong counts for each sync. OTOH, if all counters get reset to 0 after calling `Pings.historySync.send()` or `Pings.bookmarksSync.send()` (even if the ping fails to submit), maybe I can drop that and just use `add()`?

That's the default behaviour: if no `lifetime` is specified in the `metrics.yaml` file, we assume `ping` lifetime. Metrics with that lifetime are cleared when they are sent with the ping they are associated with. So, yes, you can just call `add` and get rid of the `CounterMetricType` changes. I would be very explicit about the `lifetime` in the metrics file, I'd add `lifetime: ping` to all the metrics to make it clear that you're relying on that behaviour.